client/src/main/java/com/battleships/connection/commands/server/commands/ShotMessageReceived.java

```diff
@@ -14,16 +21,40 @@ public ShotMessageReceived(Object value) {
     @Override
     public void execute() {
         BoardService boardService = Connection.INSTANCE.boardService;
-
         Shot shot = (Shot) value;
-        //TODO if hit, then receiver is set inactive and unready, assume always missed to show that turn changes
-        boolean missed = true;
-        Platform.runLater(() -> Connection.INSTANCE.setPlayerActive(missed));
-        Platform.runLater(() -> Connection.INSTANCE.setPlayerReady(missed));
+        boolean hitSuccessful = boardService.verifyShot(shot);
+
+        if (hitSuccessful) {
+            sendReplyToOpponentThatShotWasSuccessful(boardService, shot);
+            afterFleetSunkSendMessageToTheOpponent(boardService);
+        } else {
+            activatePlayerTurnOnOpponentMiss();
+        }
+
+    }
+
+    private void sendReplyToOpponentThatShotWasSuccessful(BoardService boardService, Shot shot) {
+        List<RawBoardField> hitCoordinates = boardService.getHitMastsCoordinates(shot);
+        Connection.INSTANCE.sendToServer(new Message(CommandType.HIT, hitCoordinates));
+    }
 
-        boardService.markButtonsAsHit(shot.getCoordinates());
+    private void afterFleetSunkSendMessageToTheOpponent(BoardService boardService) {
+        if (boardService.isFleetSunk()) {
+            Connection.INSTANCE.sendToServer(new Message(CommandType.FLEET_SUNK, ""));
+            Platform.runLater(() -> showLostAlert());
+        }
+    }
+
+    private void showLostAlert() {
+        Alert alert = new Alert(Alert.AlertType.INFORMATION);
+        alert.setHeaderText(null);
```

Comment:
I think it is not necessary to use this setHeader text. You can check it.

Reply:
Also i would do refactor for this like prepare alert and then show it.

Reply:
using alert.setHeaderText(null); allows us not to show the default header.
refactor is a good idea